GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/PackfileMaintenanceStepTests.cs

```diff
@@ -0,0 +1,123 @@
+        }
+
+        private string GitObjectRoot => this.Enlistment.GetObjectRoot(this.fileSystem);
+        private string PackRoot => this.Enlistment.GetPackRoot(this.fileSystem);
+
+        [TestCase]
+        public void ExpireRepackExpire()
+        {
+            this.GetPackSizes(out int packCount, out long maxSize, out long totalSize);
+
+            // We should have at least two packs:
+            // 1. the pack-<hash>.pack from clone.
+            // 2. a prefetch-<timestamp>-<hash>.pack from prefetch.
+            packCount.ShouldEqual(2, message: "Incorrect packfile layout for expire test");
+
+            // Ensure we have a multi-pack-index (not created on clone)
+            GitProcess.InvokeProcess(
+                this.Enlistment.RepoRoot,
+                $"multi-pack-index write --object-dir={this.GitObjectRoot}");
+
+            List<string> packsBefore = this.GetPackfiles();
+            packsBefore.Count.ShouldEqual(packCount);
+
+            this.Enlistment.PackfileMaintenanceStep();
+            this.GetPackSizes(out packCount, out maxSize, out totalSize);
+
+            // Should expire the pack-<hash>.pack file.
+            List<string> packsBetween = this.GetPackfiles();
+
+            packsBetween.Count.ShouldEqual(1, $"incorrect number of packs after first step: {packsBetween.Count}");
+
+            Path.GetFileName(packsBetween[0])
+                .StartsWith("prefetch-")
+                .ShouldBeTrue($"packsBetween[0] should start with 'prefetch-': {packsBetween[0]}");
+            string prefetchPack = packsBetween[0];
+
+            // Create new pack(s) by prefetching blobs
+            this.Enlistment.Prefetch($"--folders {Path.Combine("GVFS", "GVFS")}");
+
+            // Create a multi-pack-index that covers the prefetch packs
+            // (The post-fetch job creates a multi-pack-index only after a --commits prefetch)
+            GitProcess.InvokeProcess(
+                this.Enlistment.RepoRoot,
+                $"multi-pack-index write --object-dir={this.GitObjectRoot}");
+
+            this.GetPackSizes(out int afterPrefetchPackCount, out maxSize, out totalSize);
+
+            // Cannot be sure, as the prefetch uses parallel threads to get multiple packs
+            afterPrefetchPackCount.ShouldBeAtLeast(2);
+
+            this.Enlistment.PackfileMaintenanceStep(batchSize: totalSize - 1);
+            this.GetPackSizes(out packCount, out maxSize, out totalSize);
+
+            packCount.ShouldEqual(afterPrefetchPackCount + 1, $"incorrect number of packs after second step: {packCount}");
+
+            // We should expire all packs except the one we just created,
+            // and the prefetch pack which is marked as ".keep"
+            this.Enlistment.PackfileMaintenanceStep();
+            List<string> packsAfter = this.GetPackfiles();
+            packsAfter.Count.ShouldBeAtMost(afterPrefetchPackCount - 1, $"incorrect number of packs after third step: {packCount}");

```

Comment:
Getting a bit lost in the math here.

Previous to the PackfileMaintenanceStep we had afterPretchPackCount +1, not we should have at most afterPrefectCount -1, i.e. two packs should be gone?  Can you explain?

Reply:
This `ShouldBeAtMost()` statement is out-dated! Now that we are able to specify `totalSize - 1`, we can ensure that we create one pack that contains all of the other packed objects. BUT we keep the most-recent prefetch pack (marked with ".keep"). Thus, we should be able to enforce _exactly two_ packs.